### PR TITLE
[test] Add rom_e2e_bootstrap_phase1 tests

### PR DESCRIPTION
**Pull request by alphan**
This PR adds e2e tests for the first phase of bootstrap:
* #14459
* #14460
* #14461 
* #14462 

This PR also fixes 2 issues that I ran into while adding these e2e tests:
* #14871
* #14873

### CODE REVIEW

File: sw/device/silicon_creator/rom/bootstrap.c [hunk 1]
```diff
@@ -179,16 +179,20 @@ static rom_error_t bootstrap_page_program(uint32_t addr, size_t byte_count,
...
     size_t word_count =
         (kFlashProgPageSize - prog_page_misalignment) / sizeof(uint32_t);
+    if (word_count > rem_word_count) {
+      word_count = rem_word_count;
+    }
     err_0 = flash_ctrl_data_write(addr, word_count, data);
     rem_word_count -= word_count;
     data += word_count * sizeof(uint32_t);
-    addr &= ~kFlashProgPageMask;
+    addr += word_count * sizeof(uint32_t);
```
Comment (a-will) [inline sw/device/silicon_creator/rom/bootstrap.c:195]: This is not correct. A PAGE_PROGRAM command modifies only a single page (hence the name ;) ).

**Review by a-will — CHANGES_REQUESTED** [review]

File: sw/device/silicon_creator/rom/bootstrap.c (hunk 1 above)
Comment (alphan) [reply sw/device/silicon_creator/rom/bootstrap.c:195]: Ah man, I was totally confused! Thanks for catching! A comment would be nice here hehe

Comment (alphan) [reply sw/device/silicon_creator/rom/bootstrap.c]: Done

**Review by timothytrippel — APPROVED** [review]

**Comment by alphan** [comment]
Merging now to write phase2 tests, PLMK if you have any additional comments. Thanks!

**Pull request merged**